sky/spot/recovery_strategy.py

```diff
@@ -251,6 +251,28 @@ class FailoverStrategyExecutor(StrategyExecutor, name='FAILOVER', default=True):
 
     _MAX_RETRY_CNT = 240  # Retry for 4 hours.
 
+    def __init__(self, cluster_name: str, backend: 'backends.Backend',
+                 task: 'task_lib.Task', retry_until_up: bool) -> None:
+        super().__init__(cluster_name, backend, task, retry_until_up)
+        # Note down the cloud/region of the launched cluster, so that we can
+        # first retry in the same cloud/region. (we cannot use handle, as it
+        # can be None if the cluster is preempted)
+        self.launched_cloud_region = None
+
+    def launch(self) -> Optional[float]:
```

Comment:
Should we record this state after a successful _launch() instead? Since _launch() is called in recover().

Reply:
Good catch! Fixed. Thanks!